modules/serverpod_auth/serverpod_auth_server/lib/src/business/users.dart

```diff
@@ -125,6 +124,18 @@ class Users {
       await AuthConfig.current.onUserUpdated!(session, userInfo);
     }
 
+    // Notify if any scopes are revoked for user.
+    if (removedScopes.isNotEmpty) {
```

Comment:
Suggestion to replace comment with descriptive variable name, e.g:
```suggestion
    var scopesHaveBeenRevoked = removedScopes.isNotEmpty;
    if (scopesHaveBeenRevoked) {
```
Then it' pretty clear why the rest is happening.